pandas/core/indexers/objects.py

```diff
@@ -341,18 +342,21 @@ def get_window_bounds(
             )
             start = start.astype(np.int64)
             end = end.astype(np.int64)
-            # Cannot use groupby_indicies as they might not be monotonic with the object
+            assert len(start) == len(
+                end
+            ), "these should be equal in length from get_window_bounds"
+            # Cannot use groupby_indices as they might not be monotonic with the object
             # we're rolling over
-            window_indicies = np.arange(
-                window_indicies_start, window_indicies_start + len(indices)
+            window_indices = np.arange(
+                window_indices_start, window_indices_start + len(indices)
```

Comment:
do we not do these in int64 in the first place?

Reply:
OT do we *keep* these as int64 throughout? (we generally use np.intp for indexers elsewhere, but IIRC this was change specifically here).

cc @mroeschke

Reply:
We started with `int64` but IMO I think all these indexers should return `intp` in the future